src/fsharp/FSharp.Build/FSharpEmbedResXSource.fs

```diff
@@ -0,0 +1,144 @@
+
+    let boilerplateGetObject = "    let GetObject(name:System.String) : System.Object = ResourceManager.GetObject(name, Culture)"
+
+    let generateSource (resx:string) (fullModuleName:string) (generateLegacy:bool) (generateLiteral:bool) =
+        try
+            let printMessage = printfn "FSharpEmbedResXSource: %s"
+            let justFileName = Path.GetFileNameWithoutExtension(resx)
+            let namespaceName, moduleName =
+                let parts = fullModuleName.Split('.')
+                if parts.Length = 1 then ("global", parts.[0])
+                else (String.Join(".", parts, 0, parts.Length - 1), parts.[parts.Length - 1])
+            let generateGetObject =
+                match _targetFramework with
+                | "netstandard1.0"
+                | "netstandard1.1"
+                | "netstandard1.2"
+                | "netstandard1.3"
+                | "netstandard1.4"
+                | "netstandard1.5"
+                | "netstandard1.6" -> false // these targets don't support the `ResourceManager.GetObject()` method
+                | _ -> true // other supported runtimes, do
+            let sb = StringBuilder().AppendLine(String.Format(boilerplate, namespaceName, moduleName, justFileName))
+            if generateGetObject then sb.AppendLine(boilerplateGetObject) |> ignore
+            let sourcePath = Path.Combine(_outputPath, justFileName + ".fs")
+            printMessage <| sprintf "Generating: %s" sourcePath
+            let body =
+                let xname = XName.op_Implicit
+                XDocument.Load(resx).Descendants(xname "data")
+                |> Seq.fold (fun (sb:StringBuilder) (node:XElement) ->
+                    let name =
+                        match node.Attribute(xname "name") with
+                        | null -> failwith "Missing resource name"
+                        | attr -> attr.Value
+                    let docComment =
+                        match node.Elements(xname "value").FirstOrDefault() with
+                        | null -> failwith <| sprintf "Missing resource value for '%s'" name
+                        | element -> element.Value.Trim()
+                    let identifier = if Char.IsLetter(name.[0]) || name.[0] = '_' then name else "_" + name
+                    let commentBody =
+                        XElement(xname "summary", docComment).ToString().Split([|"\r\n"; "\r"; "\n"|], StringSplitOptions.None)
+                        |> Array.fold (fun (sb:StringBuilder) line -> sb.AppendLine("    /// " + line)) (StringBuilder())
+                    // add the resource
+                    let accessorBody =
+                        match (generateLegacy, generateLiteral) with
+                        | (true, true) -> sprintf "    [<Literal>]\n    let %s = \"%s\"" identifier name
+                        | (true, false) -> sprintf "    let %s = \"%s\"" identifier name // the [<Literal>] attribute can't be used for FSharp.Core
```

Comment:
Can we not automatically detect if we're generating for FSharp.Core?

Reply:
Not directly.  This generator only has access to whatever is in the project file, and while we could detect `FSharp.Core.fsproj` by either `$(ProjectName)` or the existence of the `FSHARP_CORE` constant, these are merely conventions that could change and it seems wrong to assume every project named `FSharp.Core` or every project with a defined constant `FSHARP_CORE` is _our_ FSharp.Core.